driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -224,6 +226,11 @@ private void setFinalResult(
                 response.getCustomPayload());
       }
       callback.onSet(connection, response, info, statement, System.nanoTime() - startTime);
+      // if the response from the server has warnings, they'll be set on the ExecutionInfo. Log them
+      // here, unless they've been disabled.
+      if (logger.isWarnEnabled() && response.warnings != null && !response.warnings.isEmpty()) {
```

Comment:
This is a bit unrelated, but why aren't the warnings set here on the `ExecutionInfo` object? This seems like the most logical place to do it.

Reply:
The reason I am not using the ExectionInfo instance here is that the warnings may not have been set on it at this point. It seems that ExecutionInfo instance is updated with the warnings during processing of the callback. If I want to use the ExecutionInfo directly here, I believe I would have to wait for the callback Future to complete.